lib/modules/manager/gradle-wrapper/artifacts.ts

```diff
@@ -53,6 +62,50 @@ async function getDistributionChecksum(url: string): Promise<string> {
   return body;
 }
 
+export async function updateBuildFile(
+  localGradleDir: string,
+  wrapperProperties: Record<string, string | undefined | null>
+): Promise<string> {
+  let buildFileName = join(localGradleDir, 'build.gradle');
+  if (!(await localPathExists(buildFileName))) {
+    buildFileName = join(localGradleDir, 'build.gradle.kts');
+  }
+
+  let buildFileContent = await readLocalFile(buildFileName, 'utf8');
+  if (!buildFileContent) {
+    logger.debug('build.gradle or build.gradle.kts not found');
+    return buildFileName;
+  }
+
+  for (const [propertyName, newValue] of Object.entries(wrapperProperties)) {
+    if (!newValue) {
+      continue;
+    }
+
+    const query = q.tree({
+      type: 'wrapped-tree',
+      maxDepth: 1,
+      search: q
+        .sym<Ctx>(propertyName)
+        .op('=')
+        .str((ctx, { value, offset }) => {
+          buildFileContent = replaceAt(
+            buildFileContent!,
```

Comment:
why the `!` is needed? 🤔

Reply:
I have no idea. The `if (!buildFileContent) {...` clause after `await readLocalFile()` should already assert that `buildFileContent` is not null at this point. Still, if I don't use `!`, I get:

```
artifacts.ts:93:13 - error TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.
```

Reply:
probably the replaceAt returns nullable type again

Reply:
Hmm no, `replaceAt` returns strictly `string` :confused:

A workaround seems to be to make `buildFileContent` a `const` and to continue with a helper variable then:
```ts
const buildFileContent = await readLocalFile(buildFileName, 'utf8');
if (!buildFileContent) {
  logger.debug('build.gradle or build.gradle.kts not found');
  return buildFileName;
}

let buildFileUpdated = buildFileContent;
...
```

Please let me know in case you'd prefer this over `!`

Reply:
yes, let's use a new variable